woocommerce/wc-admin
# Fix for translation chunks not generated on plugin activation

Fixes https://github.com/woocommerce/woocommerce-admin/issues/4615 and https://github.com/woocommerce/woocommerce-admin/issues/6022.

Previously, combined JSON translation chunks are only generated on manually updating translation in WordPress' Updater via listening to `upgrader_process_complete` hook. However, when a plugin is installed **after** a site is set to another language, WordPress does not run `upgrader_process_complete` hook despite downloading the translation files after plugin installation (Correct me if I'm wrong, but I think it's because the plugin is not activated yet). As such, there will be cases where translated texts in JavaScript does not get translated.

This PR attempts to fix translation chunks not generated by listening to `activated_plugin` hook and combines the required chunks.

Requires https://github.com/woocommerce/woocommerce/pull/28698 to be merged in WooCommerce in order for the fix to work when WooCommerce Admin is an embedded package.

### Screenshot
![image](https://user-images.githubusercontent.com/3747241/103969092-20181b00-51a0-11eb-9c9f-fc0856d6cba0.png)

### Steps to test
1. Create a test zip from this branch `npm run test:zip`
2. In a fresh JN site, set language to Español in `Settings > General`
3. Install & activate WooCommerce plugin
4. Upload zip plugin, install & activate
5. Go to WooCommerce store profiler screen and see that it is translated to Spanish
